x-pack/plugins/ml/public/embeddables/anomaly_explorer/anomaly_explorer_initializer.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiFieldNumber,
+  EuiFieldText,
+  EuiModal,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AnomalyExplorerEmbeddableInput } from '..';
+import { DEFAULT_MAX_SERIES_TO_PLOT } from '../../application/services/anomaly_explorer_charts_service';
+
+const MAX_SERIES_ALLOWED = 48;
+export interface AnomalyExplorerInitializerProps {
+  defaultTitle: string;
+  initialInput?: Partial<Pick<AnomalyExplorerEmbeddableInput, 'jobIds' | 'maxSeriesToPlot'>>;
+  onCreate: (props: { panelTitle: string; maxSeriesToPlot?: number }) => void;
+  onCancel: () => void;
+}
+
+export const AnomalyExplorerInitializer: FC<AnomalyExplorerInitializerProps> = ({
+  defaultTitle,
+  initialInput,
+  onCreate,
+  onCancel,
+}) => {
+  const [panelTitle, setPanelTitle] = useState(defaultTitle);
+  const [maxSeriesToPlot, setMaxSeriesToPlot] = useState(
+    initialInput?.maxSeriesToPlot ?? DEFAULT_MAX_SERIES_TO_PLOT
+  );
+
+  const isPanelTitleValid = panelTitle.length > 0;
+
+  const isFormValid = isPanelTitleValid && maxSeriesToPlot > 0;
+  return (
+    <EuiModal initialFocus="[name=panelTitle]" onClose={onCancel}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <FormattedMessage
+            id="xpack.ml.anomalyExplorerChartsEmbeddable.setupModal.title"
+            defaultMessage="Anomaly explorer charts configuration"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <EuiForm>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.anomalyExplorerChartsEmbeddable.panelTitleLabel"
+                defaultMessage="Panel title"
+              />
+            }
+            isInvalid={!isPanelTitleValid}
+          >
+            <EuiFieldText
+              data-test-subj={'panelTitleInput'}
+              id="panelTitle"
+              name="panelTitle"
+              value={panelTitle}
+              onChange={(e) => setPanelTitle(e.target.value)}
+              isInvalid={!isPanelTitleValid}
+            />
+          </EuiFormRow>
+
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.anomalyExplorerChartsEmbeddable.maxSeriesToPlotLabel"
+                defaultMessage="Max series to plot"
```

Comment:
Nit -  but we probably have space in the modal for `Maximum number of series to plot` here

Reply:
Updated here [`cb43b45` (#94396)](https://github.com/elastic/kibana/pull/94396/commits/cb43b4560e8dc54fe35aa2a9efffdd11b5c559e3)